crates/biome_js_parser/src/syntax/module.rs

```diff
@@ -317,47 +320,42 @@ fn parse_import_clause(p: &mut JsParser) -> ParsedSyntax {
     }
 }
 
-/// Parses the rest of an import named or default clause.
+// test js import_default_clauses
+// import e, { f } from "b";
+
+// import g, * as lorem from "c";
+/// Parses the rest of a default clause or default named clause.
 /// Rest meaning, everything after `type binding`
-fn parse_import_default_or_named_clause_rest(
+fn parse_import_default_clauses_rest(
     p: &mut JsParser,
     m: Marker,
     is_typed: bool,
 ) -> CompletedMarker {
-    match p.cur() {
-        T![,] | T!['{'] => {
+    let syntax_type = match p.cur() {
+        T![,] | T!['{'] | T![*] => {
```

Comment:
This change is responsible for the behavior change discussed [here](https://github.com/biomejs/biome/pull/1163#discussion_r1425540906)